### PR TITLE
Added Review Page

### PR DESCRIPTION
**Pull request by celestinebuendia**
- Added a Rating folder to hold Rating page contents
- Created a Rating page with components in Rating.tsx and Rating-components.tsx
- Styled to look like Figma UI in Rating.css
- Made interact-able entry fields (Rating selection fields and comment field)
- Added an API file that adds an entry to the Rating table in the database
- Made the Submit button submit review contents to database
- Added simple Review button to Course pages (otherwise the page would not be rendered in Vercel w/o a navigation route)

Testing
- Saw it run in both Vercel and npm start, resizing the window to check resizing capability
- Checked the MySQL database to see that 'Submit' successfully adds the correct content to it

### CODE REVIEW

File: app/src/App/App.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import {ClassList} from '../Directories/Directory';
 import {LoginPage} from '../Login/LoginPage';
 import {SignupPage} from '../Login/SignupPage';
 import {CoursePage} from '../CoursePages/CoursePage';
+import { ReviewPage } from '../Rating/Rating';
```
Comment (jn593) [inline app/src/App/App.tsx:8]: Remove spacing

File: app/src/App/App.tsx [hunk 2]
```diff
@@ -24,6 +25,7 @@ function App() {
           <Route path={'/login'} element={<LoginPage/>}/>
           <Route path={'/signup'} element={<SignupPage/>}/>
           <Route path={'/course/:classNum'} element={<CoursePage/>}/>
+          <Route path={'/course/:classNum/rate'} element={<ReviewPage/>}/>
```
Comment (jn593) [inline app/src/App/App.tsx:28]: Maybe change /rate to review or reviews to have the words consistent

File: app/src/Homepage/Homepage.tsx [hunk 3]
```diff
@@ -53,11 +53,11 @@ export const RatingBox: React.FC<{label: string, rating: string}> = ({ label, ra
             {label}
         </div>
     );
-  };
```
Comment (jn593) [inline app/src/Homepage/Homepage.tsx:56]: It seems that the changes here are the exact same? Try to revert the changes if they are exactly the same?

File: app/src/Rating/Rating-components.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
```
Comment (jn593) [inline app/src/Rating/Rating-components.tsx:1]: Remove spacing

File: app/src/Rating/Rating.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                    <RatingDesc rating={2}/>
+                    <RatingDesc rating={3}/>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+export interface ReviewState {
+    reviewer : string;
```
Comment (jn593) [inline app/src/Rating/Rating.tsx:43]: Is it supposed to have this spacing?

File: app/src/Rating/Rating.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            const response = await fetch('/api/PostReview', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(ratingContents),
+            });
+
+            const result = await response.json();
+            console.log(result);
```
Comment (jn593) [inline app/src/Rating/Rating.tsx:84]: Should we be posting results to the console? I feel like we shouldn't since it would expose information that the user doesn't need to see

File: app/src/Rating/Rating.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        } catch (error) {
+            console.error('Error posting review:', error);
```
Comment (jn593) [inline app/src/Rating/Rating.tsx:86]: Maybe use a different way that can explicitly show the user failed to submit? Look into alert() for the error message.

**Review by jn593 — CHANGES_REQUESTED** [review]

File: app/src/App/App.tsx (hunk 1 above)
Comment (celestinebuendia) [reply app/src/App/App.tsx:8]: Removed

File: app/src/App/App.tsx (hunk 2 above)
Comment (celestinebuendia) [reply app/src/App/App.tsx:28]: Thought it should keep with the website name, but makes sense. Changed.

File: app/src/Homepage/Homepage.tsx (hunk 3 above)
Comment (celestinebuendia) [reply app/src/Homepage/Homepage.tsx:56]: These changed removed inconsistent/unneeded spacing. The end line of RatingBox was indented for no reason. The last RatingDescProps was indented for no reason.

File: app/src/Rating/Rating-components.tsx (hunk 4 above)
Comment (celestinebuendia) [reply app/src/Rating/Rating-components.tsx:1]: Removed

File: app/src/Rating/Rating.tsx (hunk 5 above)
Comment (celestinebuendia) [reply app/src/Rating/Rating.tsx:43]: Sorry, style habit. Removed.

File: app/src/Rating/Rating.tsx (hunk 6 above)
Comment (celestinebuendia) [reply app/src/Rating/Rating.tsx:84]: Removed

File: app/src/Rating/Rating.tsx (hunk 7 above)
Comment (celestinebuendia) [reply app/src/Rating/Rating.tsx:86]: Changed to alert() message + no longer navigates back to course page on error

**Review by jn593 — APPROVED** [review]

**Review by jeremynguyen593 — APPROVED** [review]
